src/BPEssentials/Core.cs

```diff
@@ -119,9 +119,10 @@ public void ReadConfigurationFiles()
             CustomCommandsReader.ReadAndParse();
         }
 
-        public void OnReloadRequest()
+        public async void OnReloadRequestAsync()
```

Comment:
An `async` method with `void` as return is not recommended. Consider changing it to `Task`. Might not be possible in current scenario, let me know.

```suggestion
        public async Task OnReloadRequestAsync()
```

Reply:
This is not possible due to `new Action()` only accepting void.

Reply:
Alright. We'll figure this out later on.